src/api/web3.js

```diff
@@ -0,0 +1,85 @@
+import { Deployer } from '@noblocknoparty/contracts'
+import Web3 from 'web3'
+
+import { DEPLOYER_CONTRACT_ADDRESS, NETWORK } from '../config'
+
+let web3
+export let networkError
+
+function getWeb3() {
+  if (!web3) {
+    return setupWeb3()
```

Comment:
`setupWeb3()` returns a promise. I think that `getWeb3()` should just return `web3`. Clients calling should take appropriate action if it's not set.

I would rewrite this as:

```js
if (!web3) {
  setupWeb3()
}
return web3
```